sandbox-web/src/main/java/com/exadel/sandbox/team5/OrderRestController.java

```diff
@@ -53,4 +59,22 @@ public OrderDto create(@RequestBody CreateOrder createOrder) {
         }
         return orderService.createOrder(createOrder);
     }
+
+    @ApiOperation("Generating QR code")
+    @GetMapping(value = "/validate/{uuid}", produces = MediaType.IMAGE_PNG_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public byte[] generateQRCode(@PathVariable String uuid) {
+        return qrCodeService.generateQRCode(uuid);
+    }
+
+    @ApiOperation("Checks link if uuid exists in database and not expired promocode valid")
+    @GetMapping(value = "/validate/{uuid}/{expirationTime}")
+    public String validateQRCode(@PathVariable String uuid, @PathVariable String expirationTime) {
```

Comment:
We have expiration time in DB, don't need to add additional param for it
1) Find order in database by uuid
2) Validate expiration time, move this logic into service layer

Reply:
Moved to service layer, deleted expirationTime from url and added check. I also added ```promocode``` param